src/category_theory/abelian/ext.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Group.basic
+import algebra.category.Module.abelian
+import category_theory.derived
+import category_theory.linear.yoneda
+import category_theory.abelian.opposite
+import category_theory.abelian.projective
+
+/-!
+# Ext
+
+We define `Ext R C n : Cᵒᵖ ⥤ C ⥤ Module R`
+for any `R`-linear abelian category `C`
+by left-deriving in the second argument of
+the `linear_yoneda` bifunctor `(X, Y) ↦ Module.of R (unop X ⟶ Y)`
+
+## Implementation
+It's not actually necessary here to assume `C` is abelian,
+but the hypotheses, involving both `C` and `Cᵒᵖ`, are quite lengthy,
+and in practice the abelian case is hopefully enough.
+-/
+
+noncomputable theory
+
+open category_theory
+
+variables (R : Type*) [ring R] (C : Type*) [category C] [abelian C] [linear R C]
+  [enough_projectives Cᵒᵖ] -- When we have injective resolutions, just `[enough_injectives C]`
+
+/--
+`Ext R C n` is defined by left-deriving in the second argument of the `linear_yoneda` bifunctor.
+-/
+@[simps]
+def Ext (n : ℕ) : Cᵒᵖ ⥤ C ⥤ Module R :=
+functor.flip
+{ obj := λ Y, functor.left_derived ((linear_yoneda R C).obj Y) n,
+  map := λ Y Y' f, nat_trans.left_derived ((linear_yoneda R C).map f) n,
+  map_id' := λ Y, by { rw category_theory.functor.map_id, apply nat_trans.left_derived_id, },
+  map_comp' := λ Y Y' Y'' f g, by { rw functor.map_comp, apply nat_trans.left_derived_comp, } }
```

Comment:
I'm a little worried about this definition. I could just be getting confused with all the \op's floating around...

If I understand this definition correctly, given an object `Y`, we consider the functor `F := Hom(-,Y)` as a functor on C^op, and left derived it, which means, for `X : C\op`, choosing a *projective resolution* `P` of `X`, applying `F`, and taking homology. But a projective resolution of `X` in `C\op` corresponds to an injective resolution of `X.unop` in `C`, so it seems to me that this construction equivalently takes an injective resolution of the first variable in `Hom(X,Y)`, which is incorrect. Am I just confused about something simple here?

Reply:
We're considering the functor F := Hom(_,Y) as a (covariant) functor from C^op to R-mod. This functor is *left* exact, because if 0 -> A -> B in C^op then B ->>A "in reality" so Hom(A,Y) -> Hom(B,Y) is injective. Hence we should be _right_ deriving it, which means that C^op should have enough injectives, meaning C has enough projectives, so we're taking a projective resolution of the first variable, which seems Ok to me. What do you think?

Reply:
I think that's part of my point @kbuzzard -- note the assumption `[enough_projectives Cᵒᵖ]` on line 32.

Reply:
Aah, so perhaps I am just agreeing with you! I now agree that this code looks fishy.